src/test_quick_sort.py

```diff
@@ -0,0 +1,76 @@
+# _*_ encoding: utf-8 _*_
+"""Test Implementation of quicksort Algorithm."""
+
+import random
+import pytest
+
+RANDOM_INSTANCES = [random.sample(range(1000),
+                    random.randrange(2, 4)) for n in range(50)]
+
+
+@pytest.mark.parametrize("seq", RANDOM_INSTANCES)
+def test_quick_sort(seq):
+    """Test insertion sort results equal built-in python sort results."""
+    from quick_sort import quick_sort
+    assert quick_sort(seq) == sorted(seq)
+
+
+def test_sort_simple():
+    """Test on simple case."""
+    from quick_sort import quick_sort
+    a_list = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
+    assert quick_sort(a_list) == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+def test_stable():
+    """Test identical items in list retain stable position on sort."""
```

Comment:
Quick sort is not guaranteed to be stable. (http://www.sorting-algorithms.com/quick-sort)

And, is this test doing what it proposes to do? Try adding the line `assert check_list[2] is check_list[3]`. What are the implications of this?